### PR TITLE
Fixing Typescript Rules lint warnings

### LINKED ISSUE #1201: Fix Typescript Rules lint warnings
**Describe the bug**
Currently, there are too many **Typescript Rules** ESLint warnings in the **develop** branch. 

Most of them are:-
"@typescript-eslint/ban-ts-comment",
"@typescript-eslint/no-explicit-any",
"@typescript-eslint/no-inferrable-types",
"@typescript-eslint/no-non-null-asserted-optional-chain",
"@typescript-eslint/no-non-null-assertion",
"@typescript-eslint/no-var-requires",

**To Reproduce**

1. Clone the **talawa-api** repository.
2. Install the project dependencies. (npm i)
3. In .eslintrc.json, change **Typescript Rule**, from line 49 to 54, from "off" to "warn", just like this:-
pic2

4. Run **npm run lint:check**

**Expected behavior**
Lint should report 0 Warnings and 0 Errors

**Actual behavior**
Lint now reports too many warnings.

**Additional details**
Multiple contributors can work together as there are too many warnings.

### PR DESCRIPTION
**Pull request by parthiv360**
**What kind of change does this PR introduce?**
This pull request is related to typescript lint warnings. I have resolved around 15 warnings and made changes to 9 files.

**Issue Number:**
#1201 
Fixes #1201 
**Did you add tests for your changes?**
No

**Snapshots/Videos:**
N/A

**If relevant, did you update the documentation?**
No

**Summary**
I have updated some of the typescript lint warnings that came up when running npm run lint:check.

Some of the warnings that I have resolved were:
1. Removing the "specify a different type" error after going through the codebase.
2. Adding proper null-check in various cases where null-checks were missing and resulted in a warning.

**Does this PR introduce a breaking change?**
No

**Other information**

N/A

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-api/blob/master/CONTRIBUTING.md)?**

Yes

### CODE REVIEW

**Comment by parthiv360** [comment]
Hey @palisadoes @tasneemkoushar can you please review this PR and merge it? Thank you.

**Comment by palisadoes** [comment]
@tasneemkoushar Should we also edit `.eslintrc.json` to make the linting fail?

**Comment by parthiv360** [comment]
Hey @palisadoes @tasneemkoushar any update on this PR?

**Comment by palisadoes** [comment]
@parthiv360

1. please fix the conflicting files.
2. Please also edit the `.eslintrc.json` to make the future linting fail for these fixed linting situations.

**Comment by parthiv360** [comment]
Hey @palisadoes , Thank you for the review. I will make those changes. Do I need to convert the .eslintrc.json, and change the Typescript Rule, from line 49 to 54, from "off" to "warn"?

**Comment by palisadoes** [comment]
1. The aim is to make sure all our code is compliant, so we need to fail when the linting fails.
2. If there are any other `warns` in the file, these should be converted to fail too and tested. This may require another issue to be created.

**Comment by parthiv360** [comment]
Sure, Thank You

**Comment by parthiv360** [comment]
@palisadoes I have made those changes. Can you please review? Thank you.

File: .eslintrc.json [hunk 1]
```diff
@@ -63,12 +63,12 @@
...
 
     // Typescript Rules
-    "@typescript-eslint/ban-ts-comment": "off",
-    "@typescript-eslint/no-explicit-any": "off",
-    "@typescript-eslint/no-inferrable-types": "off",
-    "@typescript-eslint/no-non-null-asserted-optional-chain": "off",
-    "@typescript-eslint/no-non-null-assertion": "off",
-    "@typescript-eslint/no-var-requires": "off",
+    "@typescript-eslint/ban-ts-comment": "warn",
+    "@typescript-eslint/no-explicit-any": "warn",
```
Comment (palisadoes) [inline .eslintrc.json:67]: Will changing these to `error` ensure that the changes you made don't get reinserted by making the workflow fail? If so, then change these to `error`

**Review by palisadoes — CHANGES_REQUESTED** [review]

File: .eslintrc.json (hunk 1 above)
Comment (parthiv360) [reply .eslintrc.json:67]: Do I need to change the "warn" to "error"?

Comment (palisadoes) [reply .eslintrc.json:67]: I think so. When the inconsistencies are found the workflow should fail. We don't want a warning and no incentive for the submitter to fix the problem.

Comment (parthiv360) [reply .eslintrc.json:67]: So currently, I am unable to commit any more changes as the warnings are converted to errors and they are aborting the commit message.

**Comment by palisadoes** [comment]
The aim of the issue is to update the code base so that the undesirable linting states are no longer present. When this occurs, there will be no more warnings. 

Your goal was to eliminate the warnings, then ensure that they do not recur by  changing the config file to `error` entries.

Has this been done?

File: .eslintrc.json (hunk 1 above)
Comment (tasneemkoushar) [reply .eslintrc.json:67]: @parthiv360 if that is the case, first try to resolve those errors

**Comment by tasneemkoushar** [comment]
@parthiv360 we want `error` instead of `warn` whenever there are linting issues in the code changes so that the contributor making the changes fixes the error before pushing it to the PR.

File: .eslintrc.json (hunk 1 above)
Comment (parthiv360) [reply .eslintrc.json:67]: There are around 886 errors that are shown currently.

**Comment by parthiv360** [comment]
@tasneemkoushar So what should I do now? Do i need to remove all the 886 errors present in the codebase?

**Comment by tasneemkoushar** [comment]
@parthiv360 don't fix all 886. Fix as much as possible and change the the rule to error only for those which you fixed. Rest you can keep warn as of now

**Comment by parthiv360** [comment]
@tasneemkoushar, Yes I have tried to remove at least 15-16 errors in the code. And I have committed those changes.

**Pull request merged**